src/main/webapp/directory-page/script.js

```diff
@@ -0,0 +1,60 @@
+/** Global variables. */
+globals = {
+    school_id: 0
+}
+
+/** Functions to execute on load. */
+window.onload = function() {
+    this.getSchoolId();
+    this.getHeader();
+    this.getEvents();
+}
+
+/** Get school ID from URL parameter. */
+function getSchoolId() {
+    search_params = new URLSearchParams(window.location.search);
+    globals.school_id = search_params.get("school_id");
+}
+
+/** Update the header with the school name. */
+async function getHeader() {
+    response = await fetch(`/school?school_id=${globals.school_id}`);
+    school_info = await response.json();
+    document.getElementById("header").innerHTML = `Happening in ${school_info.name}`;
+}
+
+/** Make subject options visible when "study" is checked. */
+function changeSubjectsVisibility() {
+    if (document.getElementById("study").checked) {
+        display = 'inline';
+    }
+    else {
+        display = 'none';
+    }
+    document.getElementById("subject-header").style.display = display;
+    document.getElementById("subject-options").style.display = display;
+}
+
+/** Make sure at least one event type checked when submitting the filter. */
+function eventTypeChecked() {
+    if (!document.getElementById("study").checked && !document.getElementById("social").checked) {
+        alert("Please select at least one event type.");
+        return false;
+    }
+    return true;
+}
+
+function getEvents() {
+
+}
+
+// /** Fetches tasks from the server and adds them to the DOM. */
+// function displayEvents() {
```

Comment:
Do you plan to use this function? Same with getEvents?

Reply:
I was thinking about it and thought that since the elements on the page need to be modified, it could be easier to make the action of the form be a Javascript function that fetches from the Java servlet instead of going directly to the Java servlet and maybe having to reload the entire page.

I wasn't sure if I wanted to break it down into two functions, one that gets the parameters of the filters, if there are any, and another function that fetches from the servlet, or just have one function that does both.

Does this approach make sense, or it would be better to just directly go to the servlet when the form is submitted?

Reply:
Keep it simple for now and just fetch all events from the servlet for a particular school then store them as a list or something in your JavaScript code. Don't worry about filtering for the moment, we can add that later. Just focus on sorting by date for now.